src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmListServiceImpl.java

```diff
@@ -779,6 +790,21 @@ public boolean toggleGermplasmListStatus(final Integer listId) {
 		return this.germplasmListService.toggleGermplasmListStatus(listId);
 	}
 
+	@Override
+	public void removeGermplasmEntriesFromList(final Integer germplasmListId,
+		final SearchCompositeDto<GermplasmListDataSearchRequest, Integer> searchComposite) {
+		this.errors = new MapBindingResult(new HashMap<>(), String.class.getName());
+		final GermplasmList germplasmList = this.germplasmListValidator.validateGermplasmList(germplasmListId);
+		this.germplasmListValidator.validateListIsNotAFolder(germplasmList);
+		this.germplasmListValidator.validateListIsUnlocked(germplasmList);
+		this.searchCompositeDtoValidator.validateSearchCompositeDto(searchComposite,
+			new MapBindingResult(new HashMap<>(), String.class.getName()));
+		if (CollectionUtils.isNotEmpty(searchComposite.getItemIds()) && searchComposite.getSearchRequest() == null) {
+			this.germplasmListDataValidator.verifyListDataIdsExist(germplasmListId, new ArrayList<>(searchComposite.getItemIds()));
+		}
+		this.germplasmListService.removeGermplasmEntriesFromList(germplasmListId, searchComposite);
```

Comment:
`this.germplasmListService.removeGermplasmEntriesFromList(germplasmListId, searchComposite);` should be put inside if clause?

Reply:
It should be outside. The if clause is only to validate the itemIds if the search request is not available.